packages/protocol-contracts/contracts/ERC20Custody.sol

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: MIT
+// v1.0, 2023-01-10
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// @title ERC20Custody.
+/// @notice ERC20Custody for depositing ERC20 assets into ZetaChain and making operations with them.
+contract ERC20Custody {
+    using SafeERC20 for IERC20;
+
+    error NotWhitelisted();
+    error NotPaused();
+    error InvalidSender();
+    error InvalidTSSUpdater();
+    error ZeroAddress();
+    error IsPaused();
+    error ZetaMaxFeeExceeded();
+    error ZeroFee();
+
+    /// @notice If custody operations are paused.
+    bool public paused;
+    /// @notice TSSAddress is the TSS address collectively possessed by Zeta blockchain validators. 
+    address public TSSAddress; 
+    /// @notice Threshold Signature Scheme (TSS) [GG20] is a multi-sig ECDSA/EdDSA protocol. 
+    address public TSSAddressUpdater;
+    /// @notice Current zeta fee for depositing funds into ZetaChain.
+    uint256 public zetaFee;
+    /// @notice Maximum zeta fee for transaction.
+    uint256 immutable public zetaMaxFee;
+    /// @notice Zeta ERC20 token 
+    IERC20 immutable public zeta;
+    /// @notice Mapping of whitelisted token => true/false
+    mapping(IERC20 => bool) public whitelisted;
+    
+    event Paused(address sender);
+    event Unpaused(address sender);
+    event Whitelisted(IERC20 indexed asset);
+    event Unwhitelisted(IERC20 indexed asset);
+    event Deposited(bytes recipient, IERC20 indexed asset, uint256 amount, bytes message);
+    event Withdrawn(address indexed recipient, IERC20 indexed asset, uint256 amount);
+    event RenouncedTSSUpdater(address tssAddressUpdater);
+
+    /**
+     * @dev Only TSS address allowed modifier.
+     */
+    modifier onlyTSS() {
+        if (msg.sender != TSSAddress) {
+            revert InvalidSender();
+        }
+        _;
+    }
+
+    /**
+     * @dev Only TSS address updater allowed modifier.
+     */
+    modifier onlyTSSUpdater() {
+        if (msg.sender != TSSAddressUpdater) {
+            revert InvalidTSSUpdater();
+        }
+        _;
+    }
+
+    constructor(address TSSAddress_, address TSSAddressUpdater_, uint256 zetaFee_, uint256 zetaMaxFee_, IERC20 zeta_) {       
+        TSSAddress = TSSAddress_; 
+        TSSAddressUpdater = TSSAddressUpdater_;
+        zetaFee = zetaFee_;
+        zeta = zeta_;
+        zetaMaxFee = zetaMaxFee_;
+    }
+
+    /**
+     * @dev Update the TSSAddress in case of Zeta blockchain validator nodes churn
```

Comment:
nit: some comments you finish with `.` and some without. Let's standardize all the sentences to end with `.`